spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/support/RelaxedNames.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.support;
+
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Locale;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.util.StringUtils;
+
+/**
+ * Generates relaxed name variations from a given source.
+ *
+ * @author Phillip Webb
+ * @author Dave Syer
+ * @see RelaxedDataBinder
+ * @see RelaxedPropertyResolver
+ */
+public final class RelaxedNames implements Iterable<String> {
```

Comment:
Could you add the reason to bring over this class in here?

Reply:
Looks like I had a comment in a commit:
```
Copy over original boot's RelaxedNames as
our use case seem to be more manual rather
than actually binding. Need to revisit
this later.
```

Reply:
Can we create an issue for this.  Prob need to create issues for all `TODO` comments in the PR.